src/main/java/org/jitsi/impl/protocol/xmpp/colibri/ColibriConferenceImpl.java

```diff
@@ -943,4 +952,46 @@ public void release()
             }
         }
     }
+
+    public static class Stats {
+        private final Clock clock;
+
+        public Stats(Clock clock)
+        {
+            this.clock = clock;
+        }
+
+        public Stats()
+        {
+            this(Clock.systemUTC());
+        }
+
+        /**
+         * An average of the time it takes to make allocate channel requests
+         * to JVB.
+         */
+        private final RateStatistics allocateChannelsReqTimes =
+            new RateStatistics((int)Duration.ofMinutes(1).toMillis());
+
+        private final RateStatistics healthCheckReqTimes =
+            new RateStatistics((int)Duration.ofMinutes(1).toMillis());
+
+        /**
+         * Notify the stats object how long an allocate channels request took
+         * to execute
+         * @param nanos the time, in nanoseconds
+         */
+        void allocateChannelsRequestTook(long nanos) {
+            allocateChannelsReqTimes.update((int)nanos, clock.millis());
```

Comment:
An  `int` can only hold about 4 seconds expressed in nanos. Do we care about nanosecond precision?

Reply:
`RateStatistics` uses longs for everything  but the `update` API 🤷